paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java

```diff
@@ -208,6 +208,16 @@ public void dropTable(Identifier identifier, boolean ignoreIfNotExists)
                     identifier.getDatabaseName(), identifier.getObjectName(), true, false, true);
         } catch (TException e) {
             throw new RuntimeException("Failed to drop table " + identifier.getFullName(), e);
+        } finally {
+            // Deletes table directory to avoid schema in filesystem exists.
+            Path path = getDataTableLocation(identifier);
+            try {
+                if (fileIO.exists(path)) {
+                    fileIO.deleteDirectoryQuietly(path);
```

Comment:
If hive client fails to `delete table` and schema in filesystem is deleted, paimon cannot delete the table in hive anymore due to the validation in line 198, will it happen?

Reply:
@FangYongs, `paimonTableExists` depends on whether the table in hive exist, not the table on filesystem. Therefore, when the table in hive isn't delete, `paimonTableExists` also return true and the table in hive could also be dropped.

Reply:
Can we give more information in line 191? Users can still list tables if we delete files here. But when users query data from the table, there will be exception with message "There is no paimond in ...". Should we need to suggest users to drop the table again?

Reply:
@FangYongs, I think this again. It's better to delete table directory in filesystem after dropping hive table successfully. Because we should keep the table consistency between which in filesystem and which in Hive metastore, otherwise the `listTables` behavior is inconsistent.